cwl/src/main/scala/cwl/WorkflowStep.scala

```diff
@@ -39,9 +39,14 @@ case class WorkflowStep(
                          hints: Option[Array[CwlAny]] = None,
                          label: Option[String] = None,
                          doc: Option[String] = None,
-                         scatter: Option[String :+: Array[String] :+: CNil] = None,
+                         scatter: ScatterVariables = None,
                          scatterMethod: Option[ScatterMethod] = None) {
 
+  // We're scattering if scatter is defined, and if it's a list of variables the list needs to be non empty
```

Comment:
see above point about possible nonexistence of `ScatterFeatureRequirement` in a CWL